src/main/scala/com/gu/contentapi/firehose/kinesis/EventProcessor.scala

```diff
@@ -0,0 +1,119 @@
+package com.gu.contentapi.firehose.kinesis
+
+import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.{ IRecordProcessor, IRecordProcessorCheckpointer }
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownReason
+import com.typesafe.scalalogging.LazyLogging
+import com.twitter.scrooge.ThriftStruct
+import java.util.concurrent.atomic.{ AtomicInteger, AtomicLong }
+import java.util.{ List => JList }
+
+import com.gu.contentapi.firehose.kinesis.EventProcessor.EventWithSize
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+abstract class EventProcessor[EventT <: ThriftStruct](
+  checkpointInterval: Duration = 30.seconds,
+  maxCheckpointBatchSize: Int = 20
```

Comment:
Make these configurable? At the moment `CapiEventProcessor` doesn't set them, so they are set to their default values.